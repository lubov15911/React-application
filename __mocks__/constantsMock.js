export const moviePreviewData = {
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    vote_average: 7.8,
    tagline: 'The celebration of a lifetime',
    release_date: '2017-10-27',
    runtime: 105,
    genres: ["Adventure", "Comedy", "Family", "Animation"],
    overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz.',
};

export const filmsData = [{
    id: 354912,
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    release_date: '2017-10-27',
    genres: ["Adventure", "Comedy", "Family", "Animation"],
}, {
    id: 269149,
    poster_path: "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
    title: "Zootopia",
    release_date: "2016-02-11",
    genres: ["Animation", "Adventure", "Family", "Comedy"],
}, {
    id: 333339,
    poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
    title: 'Ready Player One',
    release_date: '2018-03-28',
    genres: ["Adventure", "Science Fiction", "Action"],
}, {
    id: 337167,
    poster_path: "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    title: "Fifty Shades Freed",
    release_date: "2018-03-28",
    genres: ["Drama", "Romance"],
}];

export const searchOptionsData = {
    title: 'Title',
    genre: 'Genre'
};

export const sortOptionsData = {
    rating: 'Rating',
    release: 'Release date'
};

export const searchValueData = 'Ready';
