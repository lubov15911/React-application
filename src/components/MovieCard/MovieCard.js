import React from 'react';
import PropTypes from 'prop-types';

import './MovieCard.scss';

const PROP_TYPES = {
    movieData: PropTypes.shape({
        poster_path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        vote_average: PropTypes.number.isRequired,
        release_date: PropTypes.string.isRequired,
        tagline: PropTypes.string.isRequired,
        runtime: PropTypes.number.isRequired,
        overview: PropTypes.string.isRequired,
    }).isRequired,
};

const MovieCard = ({ movieData, }) => {
    const { poster_path, title, vote_average, tagline, release_date, runtime, overview, } = movieData;

    return (
        <div className="movie-card">
            <img src={poster_path} alt="" className="movie-card-poster"/>
            <div className="movie-card-info">
                <div className="movie-card-info-header">
                    <h1 className="movie-card-info-title">{title}</h1>
                    <div className="movie-card-info-rating"><span>{vote_average}</span></div>
                </div>
                <p className="movie-card-info-tagline">{tagline}</p>
                <div>
                    <span className="movie-card-highlight">{release_date.slice(0, 4)}</span> year
                    <span className="movie-card-highlight">{runtime}</span> min
                </div>
                <p className="movie-card-description">{overview}</p>
            </div>
        </div>
    );
};
MovieCard.propTypes = PROP_TYPES;

export default MovieCard;
