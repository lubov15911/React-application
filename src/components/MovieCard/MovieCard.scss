.movie-card {
  display: flex;

  .movie-card-poster {
    width: 28em;
  }

  .movie-card-info {
    .movie-card-info-header {
      display: flex;

      .movie-card-info-title {
        font-size: 5em;
        font-weight: normal;
      }

      .movie-card-info-rating {
        border: 0.1em white solid;
        border-radius: 50%;
        color: #a1e66f;
        opacity: 0.8;
        font-size: 4em;
        font-weight: normal;
      }
    }

    .movie-card-info-tagline {
      font-size: 1.4em;
    }

    .movie-card-highlight {
      color: #f65261;
      font-size: 2.4em;
    }

    .movie-card-description {
      font-size: 1.7em;
    }
  }
}
