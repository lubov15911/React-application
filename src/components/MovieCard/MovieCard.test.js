import React from 'react';
import { shallow } from 'enzyme';

import MovieCard from './MovieCard';

import { moviePreviewData } from '../../../__mocks__/constantsMock';

describe('MovieCard', () => {
    it('should render correctly', () => {
        const component = shallow(<MovieCard movieData={moviePreviewData} />);

        expect(component).toMatchSnapshot();
    });
});
