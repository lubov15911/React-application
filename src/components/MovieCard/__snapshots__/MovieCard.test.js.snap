// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render correctly 1`] = `
<div
  className="movie-card"
>
  <img
    alt=""
    className="movie-card-poster"
    src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
  />
  <div
    className="movie-card-info"
  >
    <div
      className="movie-card-info-header"
    >
      <h1
        className="movie-card-info-title"
      >
        Coco
      </h1>
      <div
        className="movie-card-info-rating"
      >
        <span>
          7.8
        </span>
      </div>
    </div>
    <p
      className="movie-card-info-tagline"
    >
      The celebration of a lifetime
    </p>
    <div>
      <span
        className="movie-card-highlight"
      >
        2017
      </span>
       year
      <span
        className="movie-card-highlight"
      >
        105
      </span>
      min
    </div>
    <p
      className="movie-card-description"
    >
      Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz.
    </p>
  </div>
</div>
`;
